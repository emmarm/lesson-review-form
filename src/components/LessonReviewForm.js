import React from 'react';
import { Field, reduxForm } from 'redux-form';

import RangeInput from './RangeInput';
import RATING_FIELDS from './data/ratingFields';

const renderField = field => (
  <Field
    name={field.inputId}
    component={RangeInput}
    key={field.inputId}
    inputId={field.inputId}
    className="rating-field"
    label={field.label}
    title={field.description}
  />
);

const LessonReviewForm = () => (
  <form className="review-form">
    {RATING_FIELDS.map(field => renderField(field))}
  </form>
);

export default reduxForm({
  form: 'lessonReview'
})(LessonReviewForm);
