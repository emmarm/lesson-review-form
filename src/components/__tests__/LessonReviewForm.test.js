import React from 'react';
import { shallow } from 'enzyme';

import { LessonReviewForm } from '../LessonReviewForm';
import RATING_FIELDS from '../data/ratingFields';

describe('LessonReviewForm', () => {
  const wrapper = shallow(<LessonReviewForm />);

  it('renders LessonReviewForm snapshot correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a RangeInput for each ratingField', () => {
    const ratingFields = wrapper.find('.rating-field');
    expect(ratingFields.length).toEqual(RATING_FIELDS.length);
  });
});
