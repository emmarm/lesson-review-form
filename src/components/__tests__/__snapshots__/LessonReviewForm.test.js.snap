// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LessonReviewForm correctly 1`] = `
<form
  className="review-form"
>
  <RangeInput
    className="rating-field"
    inputId="fluency"
    key="fluency"
    label="Fluency"
    onChange={[Function]}
    value={0}
  />
  <RangeInput
    className="rating-field"
    inputId="pronun"
    key="pronun"
    label="Pronunciation"
    onChange={[Function]}
    value={0}
  />
  <RangeInput
    className="rating-field"
    inputId="comp"
    key="comp"
    label="Comprehension"
    onChange={[Function]}
    value={0}
  />
  <RangeInput
    className="rating-field"
    inputId="vocab"
    key="vocab"
    label="Active Vocabulary"
    onChange={[Function]}
    value={0}
  />
  <RangeInput
    className="rating-field"
    inputId="grammar"
    key="grammar"
    label="Active Grammar"
    onChange={[Function]}
    value={0}
  />
</form>
`;
